File: November 2024 Leetcode Solution/November-24.java
class Solution {
    public long maxMatrixSum(int[][] matrix) {
        long totalSum = 0;
        int minAbsValue = Integer.MAX_VALUE;
        int negativeCount = 0;

        for (int[] row : matrix) {
            for (int value : row) {
                int absValue = Math.abs(value);
                totalSum += absValue;
                minAbsValue = Math.min(minAbsValue, absValue);
                if (value < 0) negativeCount++;
            }
        }

        if (negativeCount % 2 != 0) {
            totalSum -= 2L * minAbsValue;
        }

        return totalSum;
    }
}

File: November 2024 Leetcode Solution/November-24.py
class Solution:
    def maxMatrixSum(self, matrix: List[List[int]]) -> int:
        total_sum = sum(abs(value) for row in matrix for value in row)
        min_abs_value = min(abs(value) for row in matrix for value in row)
        negative_count = sum(1 for row in matrix for value in row if value < 0)
        
        return total_sum - (2 * min_abs_value if negative_count % 2 != 0 else 0)
